proposals/unsigned-types.md

```diff
@@ -70,6 +70,25 @@ When one provides an external declaration for some native platform API (either C
 that declaration can contain unsigned types natively. Unsigned types in Kotlin would allow to represent such declarations
 without unwittingly altering their semantics by substituting unsigned integers with signed ones.
 
+## Non-goals
+
+### Non-negative integers
+
+While unsigned integers can only represent positive numbers and zero, it's not a goal to use them 
+where non-negative integers are required by application domain, for example, as a type of collection size or collection index value.
+
+First, even in these cases using signed integers helps to detect accidental overflows and signal error conditions, 
+such as `List.lastIndex` being -1 for an empty list or `List.indexOf` returning -1 when an element is not found in the list.
+
+Second, unsigned integers cannot be treated as a range-limited version of signed ones because their range of values 
+is not a subset of the signed integers range, i.e. neither signed, nor unsigned integers are not subtype of each other.
```

Comment:
`neither signed, nor unsigned integers are not subtype of each other.` => `neither signed, nor unsigned integers are subtypes of each other.` (double negation + plural)